### DISCUSSION TITLE
About LIBOQS and OPENSSL

### DISCUSSION THREAD

**Discussion by RideDi** [open-quantum-safe/liboqs#1612] [Q&A] [answered]
Is openssl necessary?
For example, if I only need quantum-safe algorithms instead of using hybrid signatures, do I not need openssl?

Does LIBOQS all inherit OPENSSL?or does it supplement OPENSSL with quantum-safe algorithms?

**Comment by Martyrshot** [comment, marked as answer]
OpenSSL is not necessary to use liboqs, however it is enabled by default. Depending on how liboqs is configured, it can use openssl's implementations of some primitives (such as hashing). However, for portability, liboqs does also ship with its own implementations of these primitives as well. In some cases there are assembly optimized versions of these primitives, such as for SHA2 and AES on x86 and aarch64, and in other cases the primitives implementation may be written entirely in C. In general openssl is enabled by default because they tend to have extremely optimized implementations. :)

If you want to use quantum-safe algorithms with openssl itself, you can use the [oqs-provider](https://github.com/open-quantum-safe/oqs-provider) with openssl3.

**Reply by RideDi** [reply]
I understand, thank you very much for your reply. :)
